src/pool/srv_pool.c

```diff
@@ -50,6 +50,8 @@ struct pool_svc_event {
 #define DF_PS_EVENT	"rank=%u inc="DF_U64" src=%d type=%d"
 #define DP_PS_EVENT(e)	e->psv_rank, e->psv_incarnation, e->psv_src, e->psv_type
 
+#define RECHOOSE_SLEEP_MS 500
```

Comment:
Agree with making this a constant. I wonder if the sleep is required at all, though. Certainly 500ms is better than 1s in terms of responsiveness, but could it be 250ms? Or 0?

Reply:
I agree with being more aggressive and reducing to something like 250ms. It is unclear to me whether 0 is a good idea. I'm trying to think about the possibility of, and then what happens if the pool service is in bad shape (no quorum due to losing too many replicas) and the MS engine code gets into a very tight loop constantly trying to "find the leader"? Also I'm not sure what might happen after a dRPC timeout whether this code remains actively looping in the engine.

Reply:
reran with a different PR when I realised I hadn't actually reduced it to 250ms: https://github.com/daos-stack/daos/pull/8624